pkgs/by-name/li/libjxl/package.nix

```
@@ -121,6 +124,11 @@ stdenv.mkDerivation rec {
   # the second substitution fix regex for a2x script
   # https://github.com/libjxl/libjxl/pull/3842
   postPatch = ''
+    # Make sure we do not accidentally build against some of the vendored dependencies
+    # If it asks you to "run deps.sh to fetch the build dependencies", then you are probably missing a JPEGXL_FORCE_SYSTEM_* flag
+    shopt -s extglob
+    rm -rf third_party/!(sjpeg)/
```

I like to use subshells when setting shell options to not risk disrupting any glue code / later phases and hooks / etc.
```suggestion
    (
      shopt -s extglob
      rm -rf third_party/!(sjpeg)/
    )
```